### PR TITLE
Separate Step and Sidecar types

### PR DESCRIPTION
**Pull request by Peaorl**
# Changes

This PR separates the Step and Sidecar types from each other (i.e. they are not simply aliases anymore).
This change is needed for https://github.com/tektoncd/pipeline/issues/1690 and https://github.com/tektoncd/pipeline/pull/3013.
Currently only one function relies on these two types to be aliases.
These changes are partially based on https://github.com/tektoncd/pipeline/pull/3013/files.

Additionally, .swp is added to the .gitignore file.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes
    
```release-note
NONE
```

### CODE REVIEW

**Comment by chmouel** [comment]
fyi we usually don't add code editor exclude files in gitignore but usually to your  global ignore/exclude file, see this article for more info : 
https://julien.danjou.info/properly-managing-your-gitignore/

/ok-to-test

**Comment by sbwsg** [comment]
> fyi we usually don't add code editor exclude files in gitignore

I think I understand the reason why but in this case we already have emacs, vscode and jetbrains ignore rules in place.  Should we remove those?

File: pkg/apis/pipeline/v1beta1/task_types.go [hunk 1]
```diff
@@ -125,8 +125,15 @@ type Step struct {
 	Script string `json:"script,omitempty"`
 }
 
-// A sidecar has the same data structure as a Step, consisting of a Container, and optional Script.
-type Sidecar = Step
+// Sidecar has nearly the same data structure as Step, consisting of a Container and an optional Script, but does not have the ability to timeout.
```
Comment (sbwsg) [inline pkg/apis/pipeline/v1beta1/task_types.go:128]: minor nit: suggest matching comment style with others in the file and wrapping the lines around 80-100 characters.

similarly for other comments in this pr.

**Comment by Peaorl** [comment]
/test pull-tekton-pipeline-integration-tests

(Retrying because there was a general problem fetching docker images)

File: pkg/apis/pipeline/v1beta1/task_types.go (hunk 1 above)
Comment (Peaorl) [reply pkg/apis/pipeline/v1beta1/task_types.go]: Updated in new commit!

**Comment by sbwsg** [comment]
Could you squash the commits into one?

Ah, you'll also need to include the following at the bottom of your PR description:

    ```release-note
    NONE
    ```

/kind cleanup

**Pull request merged**